### PR TITLE
os/include: Add DEBUGASSERT_INFO macro for descriptive debug output in case of DEBUGASSERT

### PR DESCRIPTION
**Pull request by amogh-samsung**
- DEBUGASSERT_INFO gives out a descriptive output similar to ASSERT_INFO in case of DEBUGASSERTs

Sample output: -
```
TASH>>Hello, World!!
up_assert: Assertion failed at file:hello_main.c line: 71 task: appmain
up_assert: This is DEBUGASSERT_INFO in hello
up_dumpstate: Code asserted in normal thread!
up_dumpstate: Current SP is User Thread SP: 20090758
```

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

### CODE REVIEW

File: os/include/assert.h [hunk 1]
```diff
@@ -103,6 +103,13 @@ extern char assert_info_str[CONFIG_STDIO_BUFFER_SIZE];
 
 #ifdef CONFIG_DEBUG
 
+#define DEBUGASSERT_INFO(f) \
+	{ if (!(f))	{ \
+			  snprintf(debugassert_info_str, CONFIG_STDIO_BUFFER_SIZE, fmt, ##__VA_ARGS__); \
```
Comment (kishore-sn) [inline os/include/assert.h:108]: `debugassert_info_str -> assert_info_str`

Comment (amogh-samsung) [reply os/include/assert.h]: Done

File: os/include/assert.h [hunk 2]
```diff
@@ -89,7 +89,7 @@ extern char assert_info_str[CONFIG_STDIO_BUFFER_SIZE];
 	{ if (!(f))	{ \
```
Comment (swasthik22) [inline os/include/assert.h:89]: can you add the sample output to commit description as well.

File: os/include/assert.h [hunk 3]
```diff
@@ -89,7 +89,7 @@ extern char assert_info_str[CONFIG_STDIO_BUFFER_SIZE];
...
 			  snprintf(assert_info_str, CONFIG_STDIO_BUFFER_SIZE, fmt, ##__VA_ARGS__); \
 			  up_assert((const uint8_t *)__FILE__, (int)__LINE__); \
-		} \
+			} \
```
Comment (sunghan-chang) [inline os/include/assert.h:92]: This macro has something wrong indentation. Let's modify it together with new commit. To split different change, could you revert this change from this commit?

File: os/include/assert.h [hunk 4]
```diff
@@ -103,6 +103,13 @@ extern char assert_info_str[CONFIG_STDIO_BUFFER_SIZE];
 
 #ifdef CONFIG_DEBUG
 
+#define DEBUGASSERT_INFO(f, fmt, ...) \
+	{ if (!(f))	{ \
+			  snprintf(assert_info_str, CONFIG_STDIO_BUFFER_SIZE, fmt, ##__VA_ARGS__); \
+			  up_assert((const uint8_t *)__FILE__, (int)__LINE__); \
+			} \
+	}
+
```
Comment (sunghan-chang) [inline os/include/assert.h:112]: It looks something strange. How about below?
```
{ \
    if ( ) { \
        snprintf(); \
        up_assert(); \
    } \
}
```

Comment (amogh-samsung) [reply os/include/assert.h:112]: I will make the new macros with this indentation and change the indentation of the old macros in a separate commit in this PR itself

File: os/include/assert.h (hunk 3 above)
Comment (amogh-samsung) [reply os/include/assert.h]: Done

**Review by sunghan-chang — APPROVED** [review]
